src/libcore/iter/range.rs

```diff
@@ -353,6 +304,23 @@ impl<A: Step> Iterator for ops::RangeFrom<A> where
     fn size_hint(&self) -> (usize, Option<usize>) {
         (usize::MAX, None)
     }
+
+    #[inline]
+    fn count(self) -> usize {
+        usize::MAX
```

Comment:
Well with debug assertions disabled `(0u8..).nth(1000)` returns `Some(232)` and by the looks of it your version will always `panic`. Ideally it would return `None`. I guess adding `panic`s in cases where the behaviour isn't properly defined isn't too bad but I'd like to see this implementation properly fixed.